lib/ui/chat/widgets/message_widget.dart

```diff
@@ -184,13 +218,38 @@ class MessageWidget extends StatelessWidget {
     textPainter.layout(maxWidth: maxWidth);
     final lines = textPainter.computeLineMetrics();
 
-    if (lines.isNotEmpty) {
-      final lastLineWidth = lines.last.width;
-      final availableWidth = maxWidth - lastLineWidth;
-      final canFitInline = lines.length == 1 && availableWidth >= (timestampWidth + minPadding);
+    if (lines.isEmpty) {
+      return const SizedBox.shrink();
+    }
 
-      if (canFitInline) {
-        return Row(
+    // Find the longest line width
+    final longestLineWidth = lines.map((line) => line.width).reduce((a, b) => a > b ? a : b);
+
+    // Check if timestamp can fit inline with the last line
+    final lastLineWidth = lines.last.width;
+    final availableWidth = maxWidth - lastLineWidth;
+    final canFitInline = lines.length == 1 && availableWidth >= (timestampWidth + minPadding);
+
+    if (canFitInline) {
+      // Single line with inline timestamp
+      if (hasMedia && mediaWidth != null) {
+        // If media exists, message should start from left, timestamp should align to media width on the right
+        return SizedBox(
+          width: mediaWidth,
+          child: Row(
+            children: [
+              Expanded(
+                child: textWidget,
+              ),
+              SizedBox(width: minPadding),
+              TimeAndStatus(message: message, context: context),
+            ],
+          ),
+        );
+      }
+      // No media, use IntrinsicWidth for normal hugging behavior
+      return IntrinsicWidth(
```

Comment:
From this widget's docs: This class is relatively expensive. Avoid using it where possible.


Please reconsider this approach if  possible.